common/scala/src/main/scala/whisk/core/entity/AuthKey.scala

```diff
@@ -17,24 +17,31 @@
 
 package whisk.core.entity
 
+import akka.http.scaladsl.model.headers.{BasicHttpCredentials, HttpCredentials}
 import spray.json._
 import spray.json.DefaultJsonProtocol._
 
+trait AuthKeyEnv {
+  val authElements: Map[String, String]
+  def toEnvironment = authElements.toJson.asJsObject
+  def getCredentials: HttpCredentials
+}
```

Comment:
Since `authElements` is used nowhere, what do you think about narrowing this interface to

```scala
trait AuthenticationCredentials {
  def toEnvironment: JsObject
  def credentials: HttpCredentials
}
```

Since the interface includes the `getCredentials` bit, I think `AuthKeyEnv` doesn't transport the meaning of the class. After all this is about the credentials being passed in.